addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -89,6 +89,87 @@ function factory(dependencies) {
+            if (!searchedText) {
+                return unlinkAll();
+            }
+            searchedText = _.escape(searchedText);
+            searchedText = searchedText.toLowerCase();
+            const filters = [
+                // authorNameFilter
+                (message, searchedText) => {
+                    return message.author &&
+                    message.author.nameOrDisplayName && 
+                    message.author.nameOrDisplayName.toLowerCase().includes(searchedText);
+                },
+                // authorNameFilter
+                (message, searchedText) => {
+                    return message.body && message.body.toLowerCase().includes(searchedText);
+                },
+                // attachmentNameFilter
+                (message, searchedText) => {
+                    for (const attachment of message.attachments) {
+                        if (attachment.name && attachment.name.toLowerCase().includes(searchedText)) {
+                            return true;
+                        }
+                    }
+                    return false;
+                },
+                // subjectFilter
+                (message, searchedText) => {
+                    return message.subject && message.subject.toLowerCase().includes(searchedText);
+                },
+                // subTypeDescriptionFilter
+                (message, searchedText) => {
+                    return message.subtype_description && message.subtype_description.toLowerCase().includes(searchedText);
+                },
+                // tracking values
+                (message, searchedText) => {
+                    const doesTrackingValueContainsSearchedText = (trackingValue) => {
+                        return trackingValue.new_value.toLowerCase().includes(searchedText) || trackingValue.old_value.toLowerCase().includes(searchedText) 
+                    }
+                    return message.trackingValues.some(doesTrackingValueContainsSearchedText);
+                },
+            ];
+            const filteredMessages = this.thread.messages.filter((message) => {
+                return filters.some((filterFn) => filterFn(message, searchedText))
+            });
+            filteredMessages.sort((m1, m2) => m1.id < m2.id ? -1 : 1);
+            return replace(filteredMessages);
+       }
+
+        /**
+        * @private
+        * @returns {Boolean}
+        */
+        _computeIsMessageSearchPending() {
+            // If there is no thread, there is no pending
+            if (!this.thread) {
+                return false;
+            }
+            // If the data has changed, it means we're done
+            if (this._filteredMessages !== this.filteredMessages) {
+                this._filteredMessages = this.filteredMessages;
+                return false;
+            } 
+            // Otherwise, if the data has not changed, we're still pending. 
+            return true;
+        }
```

Comment:
I have strong doubts this works. In reality, while the search is client side, having a loader is useless. But when it will be server side, then it will be important. And there will be a code flux to actually easily set IsMessageSearchPending without a compute.